### DISCUSSION TITLE
Ability to override a concrete classes functionality

### DISCUSSION THREAD

**Discussion by ronnyek** [AutoFixture/AutoFixture#1394] [Q&A]
What I'm hoping to find a way to do is the following. 

Create a mock of interface ICasClient, that will provide methods I've setup, and delegate to underlying CasClient for  things I didn't provide a setup for.

I realize I could change the architecture slightly so that the functioanlity I aim to override, could be defined in a different interface, and then I could mix/match/intermingle genuine concrete types and generated types... but trying to see if somethign like this is already doable with autofixture (bear with me, I'm new to autofixture and trying to understand how to accomplish different things )

**Comment by Kralizek** [comment]
As a rule of thumb, AutoFixture aims at generating data objects not fakes of services.

There are addons (glue libraries) that help integrating AF with mocking frameworks like Moq or FakeItEasy but again these glue libraries should be seen more as a way to instrument the mocking frameworks so that their mocks/fakes return objects created via AF.

**Reply by ronnyek** [reply]
I feel like that makes sense... AF is for building the fixture and registering a service etc... What is it that Customize and customizations are supposed to achieve, or are those provided by those "glue" libraries. I do have moq and I believe a AutoFixture.Moq or something in place.

**Reply by aivascu** [reply]
@ronnyek customization is a bit of an overloaded term in AutoFixture.
`Fixture.Customizations` property is a collection of factories and relays that create parts of objects or entire objects.
`Fixture.Customize<T>()` offers a builder API for configuring how a type is created and hydrated with data each time an instance is created.
`Fixture.Customize(ICustomization)` is a way of injecting encapsulated customization behavior that can alter the fixture in a multitude of ways, e.g. adding/removing specimen builders, customizing types, adding/removing behaviors, etc. You'll usually see it used when adding the glue libraries.
